src/python/system/environment.py

```diff
@@ -332,13 +333,25 @@ def get_llvm_symbolizer_path():
   """Get the path of the llvm-symbolizer binary."""
   llvm_symbolizer_path = get_value('LLVM_SYMBOLIZER_PATH')
 
-  # Use default llvm symbolizer for the following:
-  # 1. If we don't have |LLVM_SYMBOLIZER_PATH| env variable set.
-  # 2. If this build is deleted, then our own llvm symbolizer.
-  if not llvm_symbolizer_path or not os.path.exists(llvm_symbolizer_path):
-    llvm_symbolizer_path = get_default_tool_path('llvm-symbolizer')
-
-  # Make sure that llvm symbolizer binary exists.
+  if llvm_symbolizer_path and os.path.exists(llvm_symbolizer_path):
+    # Make sure that llvm symbolizer binary is executable.
+    os.chmod(llvm_symbolizer_path, 0o750)
+
+    return_code = subprocess.call(
+        [llvm_symbolizer_path, '--help'],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL)
+    if return_code == 0:
+      # llvm-symbolize works, return it.
```

Comment:
nit: log_error so we know about it?

Reply:
environment is such a self-contained module, that is why skipped adding logging dependency to avoid cycles.

Reply:
Ah right, thanks for clarifying.